### ISSUE TITLE
Paste is duplicated in the shell prompt

### ISSUE THREAD

**Issue by thhart** [ohmyzsh/ohmyzsh#11375]
### Describe the bug

When I paste from the clipboard the content is pasted 2 times. This is regardless if I use simple mark & paste by mouse or copy & paste via the clipboard.  This is not happening in a bash or previous zsh installation. It is also not related to terminal application, it happens in gnome-terminal and konsole.

### Steps to reproduce

1. Mark and Copy something in a console with CTRL-SHIFT-C
2. Paste it with CTRL-SHIFT-V

The resulted looks like in the screenshot. Foo. was the text copied and pasted.

### Expected behavior

Pasted text should appear only once.

### Screenshots and recordings

![image](https://user-images.githubusercontent.com/2632538/206228638-4de50492-a8a9-46ab-86ab-0362ca100697.png)

### OS / Linux distribution

Ubuntu 22.04

### Zsh version

5.8.1

### Terminal emulator

konsole

### If using WSL on Windows, which version of WSL

None

**Comment by mcornella** [comment]
I can't reproduce this, could you send a trace of the shell initialization? You can capture that with
```
zsh -xvlic exit &> ~/omz-debug.log
```
and post the `omz-debug.log` file.

**Comment by thhart** [comment]
It is for sure related to a mix with my prior zsh setup. I can send you private if you are really interested since it contains lot of my system details, however I will check if I use a rather clean approach instead of mixing.

**Comment by mcornella** [comment]
Hey Thomas, it would be good to know what is behind this behavior, in case it happens to someone else too. You can send a log file at logs@ohmyz.sh if you haven't figured out the culprit yet.

Otherwise feel free to close the issue.

**Comment by Nikl174** [comment]
I had the same problem, but i found a workaround.

I  uncommented this in the `.zshrc`:
```zsh
# Uncomment the following line if pasting URLs and other text is messed up.
DISABLE_MAGIC_FUNCTIONS="true"
```
and after that, the problem was gone.

I hope, this is somehow helpful for finding the underlying bug.

**Comment by mcornella** [comment]
Hi @Nikl174, please open a new issue with more details if that setting is triggering the bug. Particularly helpful will be filling in all the information for a bug, and capture a trace as in https://github.com/ohmyzsh/ohmyzsh/issues/11375#issuecomment-1345514776. As is we are still unable to reproduce the issue. Thanks!

**Comment by g0rr0g** [comment]
I can also confirm that on Mac OS Sonoma 14.5, with a fresh installation of oh-my-zsh and using the standard Terminal, I get the double paste issue too (using command-C and command-V), and @Nikl174 's solution above works.

(Apologies for not opening a new bug as requested above, but I thought it would be useful to readers to have my comment here in the same place that I arrived from Google.)

**Comment by mcornella** [comment]
Will try to reproduce on macOS.

**Comment by IJuanI** [comment]
I always thought this was a generalized bug, lots of folks in my company do experience it. That said, after disabling and re-enabling `DISABLE_MAGIC_FUNCTIONS` I'm no longer able to replicate. I also did update omz this morning, so can't rule out it being patched.

Here's the definition of that flag based on omz-debug's output:
```batch
# *-magic is known buggy in some versions; disable if so
if [[ $DISABLE_MAGIC_FUNCTIONS != true ]]; then
  for d in $fpath; do
    if [[ -e "$d/url-quote-magic" ]]; then
      if is-at-least 5.1; then
        autoload -Uz bracketed-paste-magic
        zle -N bracketed-paste bracketed-paste-magic
      fi
      autoload -Uz url-quote-magic
      zle -N self-insert url-quote-magic
    break
    fi
  done
fi
```

And here's my `omz-debug.log` generated with `zsh -xvlic exit &> ~/omz-debug.log`. This was run before tampering with `DISABLE_MAGIC_FUNCTIONS` but after updating omz. Can't confirm if the issue was replicable or not at that time.
Part 1: https://pastebin.com/rLsqBfT8
Part 2: https://pastebin.com/KFM80rfE

If I stumble uppon the bug again, I will generate a new output and share through this thread

**Comment by mcornella** [comment]
Hi Juan, thanks for the submission.

I've gone ahead and redacted the pastebins, please go ahead and also delete them. As a general rule don't post in pastebin trace logs. GitHub Issues has attachment support, but we are working on a submission alternative that doesn't publicize potentially-sensitive logs. For now, you can email us log files at logs@ohmyz.sh. I've already downloaded these ones though.

I will try to analyze them shortly, see if I can find anything of value. Thanks again!

**Comment by robbyrussell** [comment]
Sorry, everyone — this stalled on our side, not yours. @IJuanI, you sent traces and they never got looked at properly.

Before anyone spends more time on it: does this still happen for you on a current Oh My Zsh? @IJuanI mentioned it went away after an update, and it'd be useful to know whether that held.

If it is still happening, this works around it in the meantime — the value has to be exactly `true`, and it has to come before the line that sources `oh-my-zsh.sh`:

```zsh
DISABLE_MAGIC_FUNCTIONS=true
source $ZSH/oh-my-zsh.sh
```

That turns off `bracketed-paste-magic`, which is the most likely thing involved in a duplicated paste.

**Comment by IJuanI** [comment]
Hey @robbyrussell, just checked and I can no longer replicate, I'm not sure whether I've some local patch for it or not though, can't remember where I ended up with this. At the very least I don't have the disable magic functions defined locally
